### DISCUSSION TITLE
Creating Implementation Stubs

### DISCUSSION THREAD

**Discussion by imlixixi** [nasa/fprime#1310] [Q&A] [answered]
When I operated as the  tutorial there are two problems.
In the "Generating Implementation Stubs of SignalGen" part (cd fprime/Ref/SignalGen fprime-util impl):
1.[WARNING]Failed to find settings file: /Users/lihaoyu/Desktop/fprime/Ref/settings.ini
2.[ERROR] Expected CMake variable FPRIME_FRAMEWORK_PATH to be set to '/Users/lihaoyu/Desktop/fprime', was actually set to '/Users/lihaoyu/fprime'. This is usually caused by updating the settings.ini file without purging and regenerating the accompanying build cache.

**Comment by ThibFrgsGmz** [comment, marked as answer]
Don't worry about your first problem. It's just a warning that will be corrected in issue #1302.

**Reply by imlixixi** [reply]
thank u for your help but when I re-ran the command “fprime-util generate”：
-- [fpp-tools] Searching for sbt
-- [fpp-tools] sbt found at: /usr/local/bin/sbt
CMake Error at /Users/lihaoyu/Desktop/F´ Community/fprime/cmake/fpp-download/fpp.cmake:78 (message):
-- Configuring incomplete, errors occurred!
  Build step for fpp failed: 2
See also "/Users/lihaoyu/Desktop/F´ Community/fprime/Ref/build-fprime-automatic-native/CMakeFiles/CMakeOutput.log".

  Cloning into 'src'...

  HEAD is now at a8705b3e Update version number in docs

  usage: dirname path

  make[2]: *** [fpp-tools-prefix/src/fpp-tools-stamp/fpp-tools-install] Error
  1

**Reply by ThibFrgsGmz** [reply]
I had the same problem about ```fprime-util generate``` and ```fpp```, check https://github.com/nasa/fprime/discussions/1235 to solve it.
More precisely https://github.com/nasa/fprime/discussions/1235#discussioncomment-2298136

**Reply by imlixixi** [reply]
I had checked #1235 but the problem is a little different from mine(There are some differences in the error message)

**Reply by ThibFrgsGmz** [reply]
Which version of F´ do you use? Can you pull the latest version of the project?

Have you tried to install FPP manually? The problem could be in your [fpp.cmake](https://github.com/nasa/fprime/blob/devel/cmake/fpp-download/fpp.cmake) file.

**Comment by LeStarch** [comment]
It looks like you may have moved your fprime directory after running `fprime-util generate`.  If this is the case, you need to rerun it as it stores many absolute paths.

```
fprime-util purge
fprime-util generate
```
